nautobot_circuit_maintenance/handle_notifications/sources.py

```diff
@@ -209,26 +229,32 @@ def get_notifications_from_email(
                 continue
 
             logger.log_info(
-                message=f"Retrieving notifications from {email_box.source_id} for {', '.join(providers_with_email)} since {since_txt}",
-            )
-            imap_conn = IMAP(
-                service=email_box.source_type.lower(),
-                user=email_box.source_id,
-                # How to setup GMAIL APP password
-                # https://support.google.com/accounts/answer/185833
-                password=email_box._password,  # pylint: disable=protected-access
-                imap_url=email_box.url,
+                message=f"Retrieving notifications from {notification_source.alias} for {', '.join(providers_with_email)} since {since_txt}",
             )
-            rawnotification = imap_conn.receive_emails(logger, restrict_emails, since)
+
+            try:
+                imap_conn = Source.init(alias=notification_source.alias)
+                if not imap_conn:
+                    logger.log_warning(
+                        message=f"No Notification Source configuration available or supported scheme for {notification_source.alias}.",
+                    )
+                    continue
+            except ValidationError as validation_error:
+                logger.log_warning(
+                    message=f"Notification Source {notification_source.alias} is not matching class expectations: {validation_error}",
+                )
+                continue
+
+            rawnotification = imap_conn.receive_notifications(logger, restrict_emails, since)
```

Comment:
```suggestion
                source = Source.init(alias=notification_source.alias)
                if not source:
                    logger.log_warning(
                        message=f"No Notification Source configuration available or supported scheme for {notification_source.alias}.",
                    )
                    continue
            except ValidationError as validation_error:
                logger.log_warning(
                    message=f"Notification Source {notification_source.alias} is not matching class expectations: {validation_error}",
                )
                continue

            rawnotification = source.receive_notifications(logger, restrict_emails, since)
```

Also seems like a bit incorrect to have `restrict_emails` as a parameter for generic Sources.

Reply:
yes... restrict_emails or `senders` from the signature was just an optimization to not fetch ALL the emails from an inbox (the same as `since` that limits the time scope). For Gmail API I guess we could reuse both as the API will provide a way to filter emails IIRC, but for more generic notification sources, when the time comes, this data should be ignored and another way to map to providers should be implemented.
I will add a TODO note to take care of this eventually.